src/sentry/static/sentry/app/views/organizationIntegrations/integrationListDirectory.tsx

```diff
@@ -201,9 +203,21 @@ export class OrganizationIntegrations extends AsyncComponent<
   }
 
   sortIntegrations(integrations: AppOrProviderOrPlugin[]) {
-    return integrations
-      .sort((a, b) => a.name.localeCompare(b.name))
-      .sort((a, b) => this.getInstallValue(b) - this.getInstallValue(a));
+    const sortByName = (a, b) => a.name.localeCompare(b.name);
+    const sortByPopularity = (a, b) => {
+      const weightA = Number.isInteger(weights[a.slug]) ? weights[a.slug] : 1;
```

Comment:
nit: Not sure if your usage of `Number.isInteger` is to check for undefined. 
if so, you can use the `??` operator, e.g. `const weightA = weights[a.slug] ?? 1` 
But using `??` would allow floating points, non number types, and NaN I believe.

Reply:
I get the sense that this type-checking can be solved if `popularityWeights` is typed correctly. Like doing this

```
const popularityWeights: {
  [Integration.name]: number;
} = { 
  ... 
};
```
might turn out to be better than declaring it as `readonly` (and much better than the current implementation). Either way, you should refactor one thing or the other.

Reply:
For further considerations: if you're going to abstract a 1-line function (e.g. `isSortIntegrationsByWeightActive`), why not abstract this to something like `this.getPopularityWeight`? There is a `this.getInstallValue` already.